Добавляет страницу Toys

nylus-ark opened the pull request.

nylus-ark committed:
Добавляет страницу Toys

redly commented on css/style.css:
```diff
@@ -645,15 +647,113 @@ a.header-logo:active,
+	}
+
+		.product-cards-pagination a {
+			color: inherit;
```
**Итог. Кратко**
Цвет для ссылок лучше всегда указывать явно. Тут стоит писать `color: #1e0606;` (или какой он там по дизайну)

**Чуть детальнее**
Я писал уже, повторюсь еще раз. 
Значение inherit лучше использовать супер аккуратно или вообще отказаться от него

В большинстве случаев оно и так работает для элементов, поэтому указывать inherit не обязательно
Например, ты задаешь для body цвет, поэтому для всех заголовков, параграфов, списков будет именно этот цвет. Это поведение справедливо для всех элементов, у которых в браузерных стилях не определен цвет. 

У ссылок же он задан по умолчанию явно (синий), поэтому таким и остается
Когда ты указываешь `color: inherit` ты переопределяешь цвет и поэтому он становится нужным тебе – черным. 

Ссылки очень часто отличаются цветом, от цвета блока. 
Указывая цвет не явно, а через inherit ты в данном случае создаешь "сайд-эффект", когда, например, при смене цвета у product-card-pagination поменяется и цвет ссылок тоже. 

Напомню, что нужно стараться писать код так, чтобы можно было переиспользовать блоки и верстку которая получается. Если взять твой product-card-pagination и добавить в какой-нибудь другой блок, где будет явно переопределен цвет, то опять же ссылки изменятся, а не должны

Из-за того, что код будет вести себя не явно, в разных ситуациях по-разному использовать inherit небезопасно. Есть случаи когда это точно нужно и такое поведение – плюс, но не в твоем

redly commented on toys.html:
```diff
@@ -0,0 +1,254 @@
+                            <img src="img/icons/cards-btn-next.svg" alt="next page">
+                        </a>
+                    </li>
+                </ul>
```
Данная разметка ограничивает тебя в реализации нормального оформления
Лучше было бы 

```html
<div class="product-cards-pagination">
  <a href="#" title="previous page">
    <img src="img/icons/cards-btn-prev.svg" alt="previous page">
  </a>
  <ul class="product-cards-pagination-list">
    тут список страниц как у тебя
  </ul>
  <a href="#" title="next page">
    <img src="img/icons/cards-btn-next.svg" alt="next page">
  </a>
</div>
```

Когда список будет отдельно, можно будет задать ему max-width, выравнить через margin: auto и тп
В CSS описал еще подробнее чем плохо это решение

redly commented on toys.html:
```diff
@@ -0,0 +1,254 @@
+                <ul class="product-cards-pagination">
+                    <li>
+                        <a href="#" title="previous page">
+                            <img src="img/icons/cards-btn-prev.svg" alt="previous page">
```
Лучше назвать icon-prev, icon-next или icon-arrow-prev, icon-arrow-next
Такие названия будут более абстрактные и не будут выбиваться от того, где их использовать.

redly commented on toys.html:
```diff
@@ -0,0 +1,254 @@
+                </li>
+                <li class="breadcrumbs-list-item">
+                    <a href="toys.html">Toys</a>
+                </li>
```
Мы находимся уже на этой странице, поэтому ссылку на нее саму делать не нужно
Стоит заменить на span или вообще не использовать никакой тег

```html
<li class="breadcrumbs-list-item">
  <a href="toys.html">Toys</a>
</li>
```

redly commented on toys.html:
```diff
@@ -0,0 +1,254 @@
+    <header class="header">
+        <div class="container header-container">
+            <a href="index.html" class="header-logo">
+                <img src="img/icons/union.svg" alt="DLK company logo">
```
Лучше название файла заменить на logo
Стоит исправить на всех страницах и кажется, что я уже про это писал. Видимо потерялся коммент

redly commented on toys.html:
```diff
@@ -0,0 +1,254 @@
+                <ul class="product-cards-list">
+                    <li class="product-cards-item">
+                        <a href="#" class="product-cards-item-img">
+                            <img src="/img/shark.png" alt="Sharky bean bag chair cover">
```
Не нужно дублировать alt, если есть этот же контент ниже в заголовке, подписи и тп
Alt нужен, чтобы можно было понять контент изображения не смотря на него. 
Он полезен не только, когда картинка битая и не прогрузилась, но и для устройств чтения с экрана и доступности

В случае чтения с экрана твоей разметки будет 2 раза прочитано название товара. (И в целом не совсем будет понятно даже что это название товара)

Чтобы этого не происходило можно оставить alt пустым, это будет лучше дублирования. 
Или поменять подпись и указать что там изображение товара Picture of 'Sharky been bag chair cover' product

redly replied:
Если не указывать ширину / высоту, то браузер может показать и понять размеры изображения только как скачает и декодирует. На это требуется время и обычно можно заметить как верстка прыгает по высоте, когда подгружаются изображения

Чтобы этого не происходило, тем более пока когда ты верстаешь без адаптива, лучше указывать явно width / height. Это позволяет браузеру зарезервировать нужное место под изображение и быстрее его показать

redly commented on toys.html:
```diff
@@ -0,0 +1,254 @@
+                        <a href="#" class="product-cards-item-img">
+                            <img src="/img/shark.png" alt="Sharky bean bag chair cover">
+                        </a>
+                        <div class="product-cards-item-info"> <!--Стоит ли здесь задавать отдельные классы элементам?-->
```
> Стоит ли здесь задавать отдельные классы элементам?

Да. Всё ок

redly commented on toys.html:
```diff
@@ -0,0 +1,254 @@
+                            <img src="/img/brass.png" alt="Brass deco frames">
+                        </a>
+                        <div class="product-cards-item-info">
+                            <h3 class="product-cards-title">Brass deco frames</h3>
```
Обычно заголовок тоже делают ссылкой на товар, поэтому внутри h3 стоит добавить a

redly commented on css/style.css:
```diff
@@ -645,15 +647,113 @@ a.header-logo:active,
+
+		.product-cards-pagination li:nth-child(n+2):nth-child(-n+4) {
+			margin-right: 64px;
+		}
```
Этот подход весь в целом не очень хороший, потому что он ломает основной принцип верстки – поддержку переполнения

Если я добавлю побольше элементов в пагинацию (https://cln.sh/YYwqpL) то все развалится
У меня должна быть возможность добавить столько ссылок на страницы, сколько необходимо

Если ссылок будет меньше, то пагинация тоже будет смотреться плохо (https://cln.sh/zdeG6n) она не сбалансировано выровнена, должна быть по центру

Стоит задавать просто для li margin и слева, и справа
А весь список с ссылками выравнивать через margin: 0 auto и указать максимальную ширину
Стрелки сделать отдельно

redly commented on css/style.css:
```diff
@@ -645,15 +647,113 @@ a.header-logo:active,
+
+		.product-cards-pagination a:active:not(.cards-pagination-active) {
+			opacity: 25%;
+		}
```
Активный пункт не должен быть ссылкой, поэтому этот код можно будет сделать проще

redly commented on toys.html:
```diff
@@ -0,0 +1,254 @@
+                        </a>
+                    </li>
+                    <li>
+                        <a href="#" title="" class="cards-pagination-active">1</a>
```
Ссылки страниц на самих себя – это очень плохо с точки зрения SEO + по смыслу тоже. 
Не нужно давать пользователю возможность кликнуть куда-то, чтобы он перешел на ту же страницу где находится

Опять же как и с breadcrumbs стоит заменить или на span, или убрать вовсе

redly commented on css/style.css:
```diff
@@ -645,15 +647,113 @@ a.header-logo:active,
+		flex-wrap: wrap;
+		justify-content: space-between;
+		list-style: none;
+		height: 1014px;
```
Мы уже обсуждали, что высоту стоит фиксировать в очень редкий случаях, например, для каких-нибудь блоков под баннеры, где размеры жестко задаются, для слайдеров с изображениями, для изображений и тп

В твоем же случае это список товаров, который может меняться. Здесь ни в коем случае нельзя ее ограничивать

Какие проблемы из-за одной строчки:
1. https://cln.sh/GhEev5 – Если пытаюсь добавить больше товаров, вся верстка ломается
2. https://cln.sh/HR9Q3r – Если оставляю всего один товар, огромный отступ снизу. По этой же причине и min-height тут тоже не нужен

redly commented on css/style.css:
```diff
@@ -645,15 +647,113 @@ a.header-logo:active,
+
+		.product-cards-item {
+			width: 380px;
+			height: 497px;
```
Аналогично комментарию выше про высоту
Так же так как ты используешь flex для сетки и не указываешь align-items, по умолчанию колонки в строках будут одинаковой высоты, поэтому и min-height для карточек не нужен

redly commented on css/style.css:
```diff
@@ -645,15 +647,113 @@ a.header-logo:active,
+			height: 497px;
+		}
+
+		.product-cards-item:nth-child(n+4) {
+			margin-top: 20px;
+		}
```
Непонятно почему именно для каждого 4й элемента этот отступ?
Стоит задать для всех li и можно задать снизу – margin-bottom: 20px

Тогда сколько бы не было товаров отступы между ними будут корректные

redly commented on css/style.css:
```diff
@@ -645,15 +647,113 @@ a.header-logo:active,
+				flex-direction: column;
+				align-items: center;
+				justify-content: space-between;
+				height: 112px;
```
Аналогично комментарию выше

redly commented on css/style.css:
```diff
@@ -645,15 +647,113 @@ a.header-logo:active,
+		justify-content: space-between;
+		align-items: center;
+		list-style: none;
+		height: 26px;
```
Аналогично комментариям выше

redly commented on css/style.css:
```diff
@@ -645,15 +647,113 @@ a.header-logo:active,
+				flex-direction: row;
+				justify-content: center;
+				align-items: center;
+				height: 385px;
```
Вот тут как раз для ограничения изображения можно оставить высоту, но лучше заменить на min-height
Также стоит добавить стили для самих изображений, чтобы они нормально адаптировались, если кто-то добавит изображение с большой шириной / высотой

```css
.product-cards-item-img img {
  max-width: 100%;
}
```

Эти стили говорят браузеру, что изображение должно заполнить максимум отведенного ему пространства и не больше

nylus-ark committed:
Вносит правки в Toys

Так же вносит мелкие правки во весь проект

nylus-ark committed:
Добавляет страницу Blog

Так же делает для пагинации более абстрактный класс для адаптации к другим страницам.
Убирает ссылки страниц на самих себя.

nylus-ark committed:
Добавляет страницу Contact Us

nylus-ark committed:
Добавляет popup и filters

Так же делает много правок связанных с плохой гибкостью верстки, влияющей на переполнение.
В частности правки касаются свойств ограничения высоты контейнеров и настройки отступов для всех элементов.

nylus-ark committed:
Добавляет star-raiting для footer

redly commented on blog.html:
```diff
@@ -0,0 +1,277 @@
+            </ul>
+        </div>
+
+        <section class="blog-articles"> <!--Нужно ли в подобных секциях делать скрытый заголовок?-->
```
> Нужно ли в подобных секциях делать скрытый заголовок?

Да раз используешь section, то нужно
У каждой section, article должен быть заголовок

redly commented on blog.html:
```diff
@@ -0,0 +1,277 @@
+                        <a href="#" class="blog-articles-item-img">
+                            <img src="/img/blog-lego.jpg" alt="Picture with kids constructor">
+                        </a>
+                        <span>23rd August 2018</span>
```
Дату лучше разметить через time – https://developer.mozilla.org/ru/docs/Web/HTML/Element/time

redly commented on contact-us.html:
```diff
@@ -0,0 +1,212 @@
+                    </p>
+                    <span>
+                        An expert is waiting to help you!
+                    </span>
```
Блочные элементы стоит размечать через блочные теги. По макету это такой же текст как выше, просто оформлен больше, другим цветом и тп. 

**Вариант 1**
Можно либо заменить на какой-нибудь заголовок, например, h4 и оформлять через каскад `contact-us-aside h4`

```html
<div class="contact-us-aside">
    …
    <p>Please use the form to contact us with your needs. Send us a message and we'll respond as soon as possible.</p>
    <h4>An expert is waiting to help you!</h4>
</div>
```

В этом случае тоже можно добавить классы (как ниже), но в целом, блок выглядит так, как будто его наполнение будет приходить из админки как готовый html, поэтому если есть возможность оформлять через каскад, то лучше оформлять через него

**Вариант 2**
Или все оформить как параграфы / div-ы, но с разными классами. 

```html
<div class="contact-us-aside">
    <p class="contact-us-aside-text">…</p>
    <p class="contact-us-aside-text">Please use the form to contact us with your needs. Send us a message and we'll respond as soon as possible.</p>
    <div class="contact-us-aside-title">An expert is waiting to help you!</div>
</div>
```

redly commented on css/style.css:
```diff
@@ -192,23 +226,43 @@ a.header-logo:active,
+	Оставить ли здесь эти свойства?
+	По-моему никак не влияют на содержимое
+	*/
+
```
Если ни на что не влияют – убирай

redly commented on css/style.css:
```diff
@@ -192,23 +226,43 @@ a.header-logo:active,
+
+	.products-list-item:nth-child(n+5) {
+		margin-top: 20px;
+	}
```
Очень странно, что отступ сверху задается только 5-му и последующим элементам
Лучше задавать сразу элементу, например, для `product-list-item` задать сразу `margin-bottom`

Так не будет сложных селекторов
А у `.products-list` (списка) просто сделать margin-bottom меньше

redly commented on css/style.css:
```diff
@@ -220,13 +274,18 @@ a.header-logo:active,
+			/*
+			Нужно ли здесь делать max-width 
+			для img?
+			*/
```
Да, нужно
Желательно для всех изображений всегда делать max-width: 100%, это избавит от проблем при переполнении. Картинка может быть гораздо больше чем по макету, но она не должна ломать верстку

redly commented on css/style.css:
```diff
@@ -243,41 +302,57 @@ a.header-logo:active,
+			большего размера то верстка поедет.
+			Тогда здесь нужно переписать полностью код кнопки
+			и использовать другой подход?
+			*/
```
Да, нужно ограничивать

Смотри, эти иконки они по макету сразу идут такого размера, поэтому измениться могут только в случае, если поменяется дизайн. Если дизайн изменится тебе нужно будет просто экспортнуть новые иконки и заменить размер кнопки. Верстка не сломается в это время как-то сама собой

Предусматривать размеры гибче стоит там, где мы зависим от контента. 
Например, 
- меняется количество текста и верстка ломается – плохо;
- меняется подпись внутри кнопки и кнопка ломается – тоже плохо;

и так далее

Там же где изменения будут только через дизайн, если решение подразумевает что-то фиксированное, так можно и нужно делать

redly commented on css/style.css:
```diff
@@ -243,41 +302,57 @@ a.header-logo:active,
-			background: url('../img/icons/btn-previous.svg') no-repeat 50% 50%;
-			background-color: rgba(112, 187, 220, 0.2);
-			margin-right: 6px;
+			background: url('../img/icons/btn-previous.svg') no-repeat 50% 50% rgba(112, 187, 220, 0.2);
```
Так как у тебя есть общий класс `products-nav-btn`, то можно к нему задать всё, что касается фона, но без изображения. И отдельно добавить уже фоновые изображения

То есть для products-nav-btn

```css
.products-nav-btn {
    background-color: rgba(112, 187, 220, 0.2);
    background-repeat: no-repeat;
    background-position: 50% 50%;
}

.products-nav-btn-prev {
    background-image: url('../img/icons/btn-previous.svg');
}

.products-nav-btn-next {
    background-image: url('../img/icons/btn-next.svg');
}
```

redly commented on css/style.css:
```diff
@@ -302,42 +376,41 @@ a.header-logo:active,
+			/*
+			Какое свойство здесь лучше width или min-width?
+			Переполнение по разному ведет себя визуально
+			*/
```
Так как дизайнер не отрисовал, как там будет выглядеть длинный текст, давай оставим как ты сделал
min-width вполне подходит

redly commented on css/style.css:
```diff
@@ -423,14 +509,20 @@ a.header-logo:active,
+	Не нашел другого способа как без высоты контейнера закрасить его.
+	Поставил внешний отступ для обрезки снизу, при переполнении
+	вроде все ок, только у блока всего увеличивается отсуп снизу.
+	*/
```
Все слишком сложно

1. Из-за того, что у `main-bottom-container` не поставил никакое выравнивание align-items то колонки пытаются быть одинаковыми по высоте, из-за чего, тебе приходится делать какие-то хаки, типа margin-bottom: auto. Этого делать не нужно, по макету у тебя 2 разные колонки, поэтому для main-bottom-container стоит добавить `align-items: flex-start`

2. Можно оставить конечно решение через box-shadow, но можно сделать и чуть проще. margin могут быть отрицательными, поэтому можно просто для заголовков задать отрицательный margin и вынести их за размер блока. 

Плюсы отрицательных margin: 
- Ты не зависишь от фона сайта. В случае box-shadow ты явно подкладываешь белый фон, а он может быть другой у сайта, например, градиент. Сразу будут сложности
- В случае смены размеров шрифта и тп, тебе нужно будет делать изменения у двух блоков – заголовка и контейнера. Когда ты контролируешь через margin, все будет в стилях заголовка

redly commented on css/style.css:
```diff
@@ -464,29 +548,40 @@ a.header-logo:active,
+	параграфов уходит вних без background
+	Как исправить это?
+	*/
 }
```
Посмотри выше комментарии про отрицательные margin. Это исправит и проблему с картой тоже. Ей нужно задать отрицательный margin снизу. Тогда, если мы убираем карту, то ничего не ломается

У колонок очень похожие стили, не нужно было делать разные классы dlk-aside-container, contacts-aside-container. Возможно они и нужны, но большую часть стилей точно можно объединить

Ниже пример разметки и CSS
Обрати внимание, возможно в итоге в разметке останутся лишние классы, их можно будет удалить

```html
<div class="container main-bottom-container">
    <div class="dlk-aside-container main-bottom-col">
        <h2 class="title-h2 dlk-aside-title main-bottom-title">Design Life Kids</h2>
        …
    </div>
    
    <div class="contacts-aside-container main-bottom-col">
        <h2 class="title-h2 contacts-aside-title main-bottom-title">Contacts</h2>
        …

        <div class="contacts-aside-map">
            <img src="img/contacts-map.png" alt="Map">
        </div>
    </div>
</div>
```

```css
.main-bottom-col {
    background-color: #e1f0f7;
    max-width: 543px;
    padding: 0 30px;
}

.main-bottom-col-title {
    margin: -20px 0 20px; 
    /* 
        Размеры указываю примерные
        Суть в том, что заголовок тоже имеет общие стили и их было бы логичнее объединить
    */
}
```

`contacts-aside-img` лучше переименовать в `contacts-aside-map` и задать отриц margin

redly commented on css/style.css:
```diff
@@ -464,29 +548,40 @@ a.header-logo:active,
+	/*
+	Я сделал отдельный div для img,
+	на случай замены изображения
+	*/
```
Это совершенно верно, что ты объединил изображение в div, так проще контролировать все переполнения и тп + там может быть не только изображение, а судя по макету интерактивная карта

Непонятно зачем такие сложные стили для этого блока?
Зачем display: flex и всё остальное?

Достаточно overflow: hidden и max-height

Если flex нужен, чтобы убрать небольшой отступ, что появляется у img снизу, то проще задать для img display: block;

redly commented on blog.html:
```diff
@@ -0,0 +1,277 @@
+                <div class="content-pagination">
+                    <a class="content-pagination-arrow" href="#" title="icon-arrow-prev">
+                        <img src="img/icons/cards-btn-prev.svg" alt="icon-arrow-prev">
+                    </a>
```
Можно убрать title, раз внутри есть изображение с alt
Alt лучше заменить на что-то более смысловое, типа "Previous page"

redly commented on blog.html:
```diff
@@ -0,0 +1,277 @@
+                    </ul>
+                    <a class="content-pagination-arrow" href="#" title="icon-arrow-next">
+                        <img src="img/icons/cards-btn-next.svg" alt="icon-arrow-next">
+                    </a>
```
Аналогично комментарию выше

redly commented on blog.html:
```diff
@@ -0,0 +1,277 @@
+                            <img src="/img/blog-lego.jpg" alt="Picture with kids constructor">
+                        </a>
+                        <span>23rd August 2018</span>
+                        <h2 class="blog-title">
```
Этот класс логичнее было бы назвать как у изображения – `blog-articles-item-title`
`blog-title` – как будто класс, который описывает общий заголовок блога (всей страницы)

redly commented on css/style.css:
```diff
@@ -395,6 +455,32 @@ a.header-logo:active,
+		но тогда из за разных размеров логотипов отступы по краям головного контейнера не совпадут.
+		Я могу каждый логотип который находится по краям контейнера отцентровать,
+		но так и кода больше будет, а нужно ли это все?
+		*/
```
Есть несколько подходов по написанию подобного кода. 
В большинстве случаев требуется, чтобы каждый элемент был бы какой-нибудь минимальной ширины, обычно для этого есть сетка в проекте и дизайнер выставляет все по ней. В учебном проекте с этим есть небольшие проблемы, но можно взять ширину по наибольшему логотипу и отталкиваться от него

**Подходы**

**Вариант 1** 
Задавать отступ только справа и делать выравнивание через flex-start. Каждому n-элементу, который уже не помещается, отступ сбрасывать в 0

**Минусы**

В этом подходе плохо только то, что скорее всего придется перестраивать и пересчитывать отступы на разных разрешениях. Когда помещается 2 элемента в строчку, будет `:nth-child(2n) { margin-right: 0 }`, когда три `nth-child(3n) { margin-right: 0 }` и так далее

**Вариант 2**

Задавать отступы слева и справа, одинаковые и для всех элементов. Например, если между элементами 120px, то стоит задать `margin: 0 60px`. В этом случае можно использовать justify-content: space-between, чтобы увеличить отступы на разрешениях, когда условно 4 карточки уже не помещаются по ширине. Отступы будут сбалансированы, потому что задаются с двух краев
Чтобы избежать проблем когда у первого и последнего элементов в строке появляются лишние отступы слева и справа, можно задать отрицательный margin для обертки элементов, в твоем случае это ul

То есть код будет примерно таким

```css
.partners-list-wrapper {
    overflow: hidden;
}

.partners-list {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    margin: 0 -60px;
}

.partners-list li {
    margin: 0 60px 20px;
}
```

**Плюсы** 
- Отступы сбалансированы и можно добиться лучшей адаптивности (это уже темы след курса). Адаптивность – перестроения под разные разрешения

**Минусы** 
- Нужна доп. обертка, которой придется указывать overflow: hidden, чтобы на маленьких ширинах не было скролла ненужного. Если не будет overflow, то скролл появится из-за отрицательных margin
- Из-за особенностей space-between, когда в последней строчке будет например всего 2 элемента они разбегутся по разным краям. Это можно попробовать решить через `:last-child { margin-right: 0 }` 


Попробуй 2 способа, выбери сам который считаешь нужным

redly commented on css/style.css:
```diff
@@ -395,6 +455,32 @@ a.header-logo:active,
+
+	.partners-list li:nth-child(4n+4) {
+		margin-right: 0;
+	}
```
Думаю с легкостью можно заменить на nth-child(4n), чтобы и для первой строки тоже работало

redly commented on css/style.css:
```diff
@@ -395,6 +455,32 @@ a.header-logo:active,
+
+	.partners-list li:nth-child(n+5) {
+		margin-top: 20px;
 	}
```
Выше я уже писал, что лучше сразу для всех li задавать отступ снизу (или сверху), чтобы учитывать переполнение

Этот способ очень плохо себя проявит, когда будет верстать адаптивные макеты. Там количество элементов сильно меняется и придется писать много доп. кода

redly commented on css/style.css:
```diff
@@ -634,32 +698,438 @@ a.header-logo:active,
+			сбивает выравнивание элементов в нем.
+			*/
+			max-width: 18px;
+			max-height: 26px;
```
Это из-за того, что там еще добавляется небольшой отступ после img, который зависит от размера шрифта, так как img по умолчанию строчный элемент

Можно просто задать display: block для img внутри этих ссылок, должно исправить проблему

redly commented on css/style.css:
```diff
@@ -634,32 +698,438 @@ a.header-logo:active,
+
+		.content-pagination-list a:active {
+			opacity: 25%;
+		}
```
По спецификации opacity задается не в процентах, а в дроби от 0.0 до 1.0
Значения в твоем случае должны быть 0.5 и 0.25

redly commented on css/style.css:
```diff
@@ -192,23 +226,43 @@ a.header-logo:active,
-	height: 270px;
 	padding: 0;
-	margin: 0;
+	margin: 0 0 30px 0;
```
В таких случаях тоже можно использовать короткую запись, например, `margin: 0 0 30px`

`margin: 0 20px` тоже самое что `margin: 0 20px 0 20px`
`margin: 0 0 20px` тоже самое что `margin: 0 0 20px 0`
`margin: 0 10px 20px` тоже самое что `margin: 0 10px 20px 10px`

Если тебе удобнее и проще потом читать код, который записан полной записью – оставляй
Если принцип понятен, так как в других случаях ты уже пользуешься короткой записью, то лучше исправить везде

redly commented on blog.html:
```diff
@@ -0,0 +1,277 @@
+        <section class="blog-articles"> <!--Нужно ли в подобных секциях делать скрытый заголовок?-->
+            <div class="container blog-articles-container">
+                <ul class="blog-articles-list">
+                    <li class="blog-articles-item">
```
Для таких объемных карточек принято использовать <article> или что-то подобное, но не списки
В целом разницы особо никакой нет: также нужен какой-то wrapper и дальше блоки-элементы, но общепринятая разметка будет скорее section + article

И учти, что если использовать section, то нужно предусмотреть заголовок, даже если его нет по дизайну
Можно использовать div + много article

redly commented on css/style.css:
```diff
@@ -634,32 +698,438 @@ a.header-logo:active,
+		flex-direction: row;
+		justify-content: space-between;
+		flex-wrap: wrap;
+	}
```
Очень сложно опять же. 
Зачем тебе в форме флексы? Делая флекс, тебе нужно думать всегда о том, как элементы переносятся, почему они например не с новой строки и тп

Просто обычные стандартные стили (блоки) закроют тут 90% твоих потребностей по разметке формы и управлять сеткой будет проще

Например, сейчас у тебя поломается верстка, если ты просто сократишь текст в чекбоксе

redly commented on css/style.css:
```diff
@@ -634,32 +698,438 @@ a.header-logo:active,
+
+		.contact-us-input-container {
+			display: flex;
+			flex-direction: column;
```
Аналогично комментарию выше

redly commented on css/style.css:
```diff
@@ -634,32 +698,438 @@ a.header-logo:active,
+		}
+
+			.contact-us-label {
+				font-size: 14px;
```
Тут достаточно будет добавить display: block, чтобы все получилось

redly commented on css/style.css:
```diff
@@ -634,32 +698,438 @@ a.header-logo:active,
+			.contact-us-input {
+				border: 1px solid rgba(0, 0, 0, 0.2);
+				border-radius: 0;
+				min-height: 50px;
```
Для input полей лучше явно задавать высоту всегда, то есть через height: 50px

redly commented on css/style.css:
```diff
@@ -634,32 +698,438 @@ a.header-logo:active,
+					color: #1E0606;
+					font-size: 18px;
+					line-height: 24px;
+					padding-left: 18px;
```
Междустрочный интервал лучше не задавать для полей вообще, он очень сильно и по-разному работает в разных браузерах. Можно оставить по умолчанию, тогда текст всегда будет по центру (вертикально имею ввиду)

Почему padding-left задается только для placeholder, когда нужен для всего поля? Сейчас placeholder с отступом, а когда начинаешь вводить текст, все липнет к левому краю

Обычно также предусматривают и отступ справа, такой же как слева, чтобы при большом количестве текста он опять же не прилипал к границам

Для `::placeholder` логичным будет оставить только значение цвета

redly commented on blog.html:
```diff
@@ -0,0 +1,277 @@
+                <ul class="blog-articles-list">
+                    <li class="blog-articles-item">
+                        <a href="#" class="blog-articles-item-img">
+                            <img src="/img/blog-lego.jpg" alt="Picture with kids constructor">
```
Вроде бы уже говорили по поводу того, что все пути лучше делать относительными, то есть слеш перед img не нужен

redly commented on blog.html:
```diff
@@ -0,0 +1,277 @@
+                            Top 100 Fastest Growing Companies 2019 in Philadelphia, 
+                            according to Growjo’s ranking.
+                        </p>
+                    </li>
```
В целом можно сделать и списком всё и это нормально, но все-таки для таких больших элементов лучше использовать другую разметку

Например, принято такие элементы оборачивать в article 
То есть заменить `<li>` -> `<article>`

redly commented on css/style.css:
```diff
@@ -634,32 +698,438 @@ a.header-logo:active,
+			}
+
+				.blog-articles-item-img img {
+					width: 100%;
```
Лучше использовать max-width: 100%
Потому что width будет растягивать даже небольшие изображения по всей ширине, max-width же просто ограничит сильно большие

redly commented on contact-us.html:
```diff
@@ -0,0 +1,212 @@
+            <div class="container contact-us-container">
+                <div class="contact-us-form-wrapper">
+                    <form class="contact-us-form">
+                        <div class="contact-us-input-container">
+                            <label class="contact-us-label" for="name">Name</label>
+                            <input class="contact-us-input contact-us-input-name" type="text" id="name" placeholder="Ivan Ivanov">
+                        </div>
+    
+                        <div class="contact-us-input-container">
+                            <label class="contact-us-label" for="phone">Phone</label>
+                            <input class="contact-us-input contact-us-input-tel" type="tel" id="phone">
+                        </div>
```
В целом ты правильно сделал что объединил label и input под общий контейнер, чтобы удобнее контролировать отступы

Ниже я написал, что делать flex c flex-wrap для всей формы плохая идея, поэтому нужно было раскидывать по строчкам. Для Phone и Email лучше задать еще одну доп. обертку, через которую можно будет контролировать флексы. 

Дальше. Не нужно было задавать отдельные классы для каждого типа поля. Они отличаются у тебя только шириной, поэтому тебе приходится ниже потом в css явно указывать для contact-us-input-phone и contact-us-input-email ширину. Если полей станет больше, нужно будет придумывать доп. класс и писать опять css
Нет в текущем подходе возможность легко на текущих стилях сверстать любую форму с большим количеством полей. Классы должны быть более абстрактные, что поможет как раз делать формы с любым кол-вом полей

Например, подобная разметка будет более поддерживаемой и переиспользуемой, хотя изменил совсем немного

```html
<div class="contact-us-group">
    <label class="contact-us-label" for="name">Name</label>
    <input class="contact-us-field" type="text" id="name" placeholder="Ivan Ivanov">
</div>

<div class="contact-us-row">
    <div class="contact-us-group">
        <label class="contact-us-label" for="phone">Phone</label>
        <input class="contact-us-field" type="tel" id="phone">
    </div>

    <div class="contact-us-group">
        <label class="contact-us-label" for="email">Email</label>
        <input class="contact-us-field" type="email" id="email">
    </div>
</div>
```

1. Чуть поменял название классов (это не принципиально, можно не делать)
2. Добавил доп. обертку для того, чтобы сделать колонки
3. Убрал доп. классы типа -email, -phone, -name. Можно их убрать совсем и сделать, чтобы `.contact-us-field` всегда занимал 100% ширины, саму ширину полей контролировать за счет обертки. 
Второй способ, это добавить доп. классы, например, `.contact-us-field-full-width`, `.contact-us-field-half-width` через который можно задавать будет ширину

В итоге, количество полей, которые занимают 50% ширины ты можешь делать бесконечно, как и тех, что занимают полную ширину. Меняться будет только html, собственно там нужно добавлять разметку, но CSS останется неизменным

redly commented on css/style.css:
```diff
@@ -634,32 +698,438 @@ a.header-logo:active,
+					padding-left: 18px;
+				}
+
+			.contact-us-input-name {
+				width: 600px;
+			}
+
+			.contact-us-input-tel,
+			.contact-us-input-email {
+				width: 290px;
+			}
```
Выше я описал что не так с разметкой и как с помощью немного другой сократить эти стили

redly commented on css/style.css:
```diff
@@ -634,32 +698,438 @@ a.header-logo:active,
+				width: 290px;
+			}
+
+			.contact-us-textarea {
+				resize: none;
+				width: 600px;
+				min-height: 150px;
+				margin-bottom: 1px;
+			}
```
Опять же width: 600 не нужен, если последовать примеру из разметки выше
По поводу margin-bottom: 1px выглядит как ошибка дизайнера, можно убрать тоже

redly commented on css/style.css:
```diff
@@ -634,32 +698,438 @@ a.header-logo:active,
+				margin-bottom: 1px;
+			}
+
+		.contact-us-checkbox-container {
+			display: inline-block;
+			margin-bottom: 16px;
+		}
```
Не нужно делать отдельную обертку для checkbox, может использоваться такая же, как у остальных полей
Если отступ действительно снизу другой по макету, то можно задать доп. класс, через который отступ переопределить

Также непонятно почему это обертка inline-block вдруг стала.

redly commented on contact-us.html:
```diff
@@ -0,0 +1,212 @@
+                            </label>
+                        </div>
+    
+                        <div class="contact-us-button-container">
```
Этот div в итоге не используешь в оформлении, да и он не нужен

redly commented on contact-us.html:
```diff
@@ -0,0 +1,212 @@
+                        </div>
+                        
+                        <div class="contact-us-checkbox-container">
+                            <input class="contact-us-checkbox-input" type="checkbox" id="consent">
```
По умолчанию на макете я вижу, что галочка отмечена. Ты забыл указать аттрибут для этого

redly commented on index.html:
```diff
@@ -280,5 +293,50 @@ <h4 class="footer-col-title">Connect</h4>
+                    </form> 
+                </div>
+
+                <a href="#№" class="popup-close">
```
Попал лишний символ Номера, должно быть просто `href="#"`

redly reviewed (changes requested):
Помимо всех замечаний по ревью, дополнительно на что следует обратить внимание

1. Нужно свериться с макетом внимательнее. И по отступам, и просто по первоначальным состояниям. Например, макет (https://cln.sh/fbRzHb) и верстка (https://cln.sh/gnmx9l). Видишь первый пункт Delivery должен быть выделен сразу
Рейтинг не выравнен и только на главной странице
И подобное

2. Нет никаких состояний по наведению и тп. Посмотри вкладку UI, там подробно расписано как и что

По приоритетности

- Сначала все правки после ревью
- Потом отдельным Pull Request давай доделывать все состояния по наведениям и тп. Там вероятно придется даже доверстывать часть элементов, которые нужно показывать

nylus-ark committed:
Вносит правки в проект
